pkg/cmd/util/product.go

```diff
@@ -0,0 +1,50 @@
+package util
+
+import (
+	"os"
+	"path/filepath"
+)
+
+const (
+	ProductOrigin           = `Origin`
+	ProductOpenShift        = `OpenShift`
+	ProductAtomicEnterprise = `Atomic Enterprise`
+)
+
+// GetProductName chooses appropriate product for a binary name.
+func GetProductName(binaryName string) string {
+	name := filepath.Base(os.Args[0])
+	for {
+		switch name {
+		case "openshift":
+			return ProductOpenShift
+		case "atomic-enterprise":
+			return ProductAtomicEnterprise
+		default:
+			return ProductOrigin
+		}
+	}
+}
+
+func GetPlatformName(binaryName string) string {
+	switch GetProductName(binaryName) {
+	case ProductAtomicEnterprise:
+		return "Atomic Enterprise Platform"
+	case ProductOpenShift:
+		return "OpenShift Application Platform"
+	default:
+		return "Origin Application Platform"
+	}
+}
+
+func GetDistributionName(binaryName string) string {
```

Comment:
add godoc with an excerpt of the usage of the returned value so we don't change this to something that makes no grammatical sense laster